internal/services/network/virtual_network_peering_data_source.go

```diff
@@ -0,0 +1,135 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/network/2023-11-01/virtualnetworkpeerings"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.DataSource = VirtualNetworkPeeringDataSource{}
+
+type VirtualNetworkPeeringDataSource struct{}
+
+func (VirtualNetworkPeeringDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"virtual_network_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (VirtualNetworkPeeringDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"remote_virtual_network_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+
+		"allow_virtual_network_access": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"allow_forwarded_traffic": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"allow_gateway_transit": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"only_ipv6_peering_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"peer_complete_virtual_networks_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+
+		"use_remote_gateways": {
+			Type:     pluginsdk.TypeBool,
+			Computed: true,
+		},
+	}
+}
+
+func (VirtualNetworkPeeringDataSource) ModelObject() interface{} {
+	return nil
```

Comment:
We should define a struct Model for the data source and return it here. I left a comment further up with a suggestion on what the struct should look like since we typically define this at the top of the resource

```suggestion
	return &VirtualNetworkPeeringDataSourceModel
```